### DISCUSSION TITLE
Overriding entrypoint with new executable file fails in docker run from GitHub action step

### DISCUSSION THREAD

**Discussion by pantelis-karamolegkos** [community/community#33733] [Actions]
I am trying to override the entrypoint of an image with a new executable as follows

Everything runs from github actions step FWIW

```
      - name: run docker
        run: |
          ls -l
          docker run \
          -v $GOOGLE_GHA_CREDS_PATH:/credentials.json \
          -v entrypoint-new:/usr/bin/entrypoint-new \
          --entrypoint /usr/bin/entrypoint-new \
          my-image:latest
```

The file is there of course (otherwise the mount would fail) and executable (this is the output of `ls -l` command that is invoked before `docker run`

```
-rw-r--r-- 1 runner docker 2577 Sep 21 16:57 README.md
-rwxr-xr-x 1 runner docker  100 Sep 21 16:57 entrypoint-new
-rw-r----- 1 runner docker 4056 Sep 21 16:57 gha-creds-12345.json
```

The process fails:

```
 exec: "/usr/bin/entrypoint-new": permission denied: unknown.
```

Why is that?

**Comment by airtower-luna** [comment]
Here's your problem:

```
          -v entrypoint-new:/usr/bin/entrypoint-new \
```

If you want to mount a file or directory as a volume, you have to specify its absolute path. A plain name is understood as the name for a Docker-managed volume, so with your current command Docker will create a new (assuming it didn't exist before) volume named `entrypoint-new` and mount it into the container at `/usr/bin/entrypoint-new`. Result: `/usr/bin/entrypoint-new` is an empty directory, which is obviously not executable. :wink: 

If you want to get the full path without hardcoding it, you can do something like this:

```
          -v "$(realpath entrypoint-new)":/usr/bin/entrypoint-new \
```
